languages/elixir/exercises/concept/binary-matching/.docs/introduction.md

```diff
@@ -0,0 +1,52 @@
+Elixir provides an elegant syntax working with binary data as we have seen with the `<<>>` special form provided for working with _bitstrings_.
+
+## Binary data
+
+The binary type is a specialization on the bitstring type. Where bitstrings could be of any length (any number of [bits][wiki-bit]), binaries are where the number of [bits][wiki-bit] can be evenly divided by 8. That is, when working with binaries, we often think of things in terms of [bytes][wiki-bytes] (8 [bits][wiki-bit]). A [byte][wiki-bytes] can represent integer numbers from `0 - 255`. It is common to work with [bytes][wiki-byte] values in [hexadecimal][wiki-hexadecimal], `0x00 - 0xFF`.
+
+Binaries may use the bitstring special form `<<>>` to define a literal. When defining a literal, we can use integer literals (numbers greater than 255 will overflow and only the last 8 bits of the integer will be used) and string literals. By default, the `::binary` modifier is applied to the value. We can concatenate binaries with the `<>/2` operator.
+
+```elixir
+<<255>> == <<0xFF>>
+<<"Hello">> == <<72, 101, 108, 108, 111>>
+```
+
+A _null-byte_ is another name for `<<0>>`
+
+## Strings are encoded binary data
+
+Why does the bitstring special form `<<>>` allow string literals? It is because string are encoded binaries in Elixir! Strings are encoded in [UTF-8][wiki-utf8] format. That is, they are encoded in 8-bit (one-byte) chunks, which allows them to represent more than 255 characters. Therefore, in elixir, the length of a string may not be the same as its byte representation
+
+```elixir
+# This string has a string length of 5, but is made up of 7 bytes
+string = "hełło"
+String.length(string) != byte_size(string)
+
+# Even emoji strings are encoded binaries
+"👍" == <<240, 159, 145, 141>>
+```
+
+## Pattern Matching on binary data
+
+Pattern matching is even extended to binaries, and we can pattern match on a portion of binary data much like we could for a list.
```

Comment:
In all the docs that I previously wrote in this repo, I always spelled pattern matching with a dash connecting the words (pattern-matching). I guess that was wrong and I should fix them, right?

Reply:
🤷 I think it depends how you look at it.  I think if you were saying "let's look at this pattern-matching operator", hyphenating is correct because it links the adjectives before the noun.  On it's own, since it is a noun phrase, I don't think it is.